src/persist/persist.ts

```diff
@@ -41,21 +42,42 @@ export async function persist(
   );
 
   // 3. write chunks to perdag.
-  await writeFixedChunks(perdag, fixedChunks, mainHeadHash, clientID);
+  await writeFixedChunks(
+    perdag,
+    fixedChunks,
+    mainHeadHash,
+    clientID,
+    mutationIDs,
+  );
 
   // 4. fixup the memdag with the new hashes.
   await fixupMemdagWithNewHashes(memdag, mappings);
 }
 
 async function gatherTempChunks(
   memdag: dag.Store,
-): Promise<[ReadonlyMap<Hash, dag.Chunk>, Hash]> {
+): Promise<
+  [
+    ReadonlyMap<Hash, dag.Chunk>,
+    Hash,
+    {mutationID: number; lastMutationID: number},
```

Comment:
Either make this:

```ts
type _ = [
    map: ReadonlyMap<Hash, dag.Chunk>,
    hash: Hash,
    mutationID: number,
    lastMutationID: number,
]
```

or

```ts
type _ = {
    map: ReadonlyMap<Hash, dag.Chunk>,
    hash: Hash,
    mutationID: number,
    lastMutationID: number,
}
```

mixing object and array seems a bit ugly

Reply:
the benefit of tuple over object is that it compresses better. the benefit of object is that it reads better...

Reply:
done